Add time-handling (Enable wormhole in final moments, disable money throw until final moments, etc)

NotTsunami opened the pull request:
Combines #126 and #129.

GreatOldOne committed:
Auto-uses wormhole during the final moments of the game

Minutes before the end game to start using wormholes is configurable.

GreatOldOne commented:
Thanks. I did actually solve the problem, it had some extra " causing the conflict.

Tsunamical commented:
The merge conflict was actually from the crit and all text stuff. The older branch you were on had it at option2, while the latest had them at option1. It would've messed all cherry-pick attempts. I did notice the extra quotation mark too.

GreatOldOne commented:
Your PR is exactly like mine with the exception of minsLeft name and the double quote fix.

options2.appendChild(makeNumber("setMinsBeforeEndGameToGoCrazy", "Change the number of minutes before going crazy"", "25px", minsBeforeEndGameToGoCrazy, 5, 59, updateEndGameCrazy));

That unclosed double quote was causing the conflict for me.
Everything else is the same.

Tsunamical commented:
I also have the parenthesis added to the Date object in cTime in the second commit :p

GreatOldOne commented:
Yeah, good thing you found that out because Chrome didn't bitch about it for some reason.

I think it's pretty much ready to be merged, and other contributors can start using the minsLeft variable to do stuff just before the game ends.

Commit:
Simplify time logic and disable throw money when relevant

-Fix cTime Date by adding needed parenthesis
-Keep timeLeft within main loop
-Disable throw money ability until final 30 minutes

Tsunamical commented:
Updated and ready to merge.

GreatOldOne commented on autoPlay.user.js:
```diff
@@ -1230,6 +1249,22 @@ function useMaxElementalDmgIfRelevant() {
+	// Check if Wormhole is purchased
+	if (canUseItem(ITEMS.WORMHOLE) && getActiveAbilityLaneCount(ITEMS.WORMHOLE) <= 0) {
+		// Wormhole is purchased, cooled down, and needed. Trigger it.
+		advLog('Less than 30 minutes for game to end. Triggering wormholes...', 2);
```
We can concatenate here to give the proper minsLeft value.

advLog('Less than ' + minsLeft + ' minutes for game to end. Triggering wormholes...', 2);

GreatOldOne commented on autoPlay.user.js:
```diff
@@ -253,6 +260,10 @@ function disableParticles() {
 function MainLoop() {
+	cHours = cTime.getUTCHours();
+	cMins = cTime.getUTCMinutes();
+	timeLeft = 60 - cMins;
```
Since you made all those vars global, make a function to update and retrive them.

function updateCurrentTime() {
    cHours = cTime.getUTCHours();
    cMins = cTime.getUTCMinutes();
    timeLeft = 60 - cMins;
}
function isNearEndGame() {
    if (cHours == 15 && timeLeft <= minsLeft) {
        return true;
    }
    else {
        return false;
    }
}

xPaw commented:
Should it also throw money at screen in final moments for no real reason? :smiling_imp:

GreatOldOne commented:
Yeah, I don't like that idea nor that my initial PR was merged with it, since it can cause dilemas and delay something that was ready to be merged.
If I do make the changes I suggested and submit (wormhole + time left before end game), can it be merged asap (to get it ready for today's End Game)?

xPaw commented:
Closing in favor of #138
